src/libmeasurement_kit/libevent/dns_impl.hpp

```diff
@@ -53,37 +54,66 @@ class QueryContext : public NonMovable, public NonCopyable {
     }
 };
 
+struct evdns_base_deleter {
+    void operator()(evdns_base* p) {
+        constexpr int fail_requests = 1;
+        evdns_base_free(p, fail_requests);
+    }
+};
+
+struct evaddrinfo_deleter {
+    void operator()(evutil_addrinfo* p) {
+        evutil_freeaddrinfo(p);
+    }
+};
+
+using evdns_base_uptr = std::unique_ptr<evdns_base, evdns_base_deleter>;
+using evaddrinfo_uptr = std::unique_ptr<evutil_addrinfo, evaddrinfo_deleter>;
```

Comment:
> Valgrind does pass on master

`make check-am-valgrind` produces at least following failures for me both on "master" (67441ddee9bc1e6057982efa7757522844b34426) and this branch:

```
FAIL: test/mlabns/mlabns
FAIL: test/ndt/ndt_test
FAIL: test/ndt/run
FAIL: test/http/request
FAIL: test/ooni/web_connectivity
FAIL: test/net/connect
```

All these tests fail with alike assertion failures somewhere near `EC_KEY_generate_key`:
```
vex: the `impossible' happened:
   isZeroU
vex storage: T total 1462491848 bytes allocated
vex storage: P total 640 bytes allocated
valgrind: the 'impossible' happened:
   LibVEX called failure_exit().
```

That's a known bug in valgrind+openssl: [1](https://bugs.kde.org/show_bug.cgi?id=356393), [2](https://bugs.launchpad.net/ubuntu/+source/valgrind/+bug/1574437), [3](https://bugzilla.redhat.com/show_bug.cgi?id=1326024).

Other tests are OK. I was using Valgrind-3.11.0 and LibVEX